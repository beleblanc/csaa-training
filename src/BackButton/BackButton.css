105,109,112,111,114,116,32,82,101,97,99,116,32,102,114,111,109,32,39,114,101,97,99,116,39,59,13,10,10,99,111,110,115,116,32,66,97,99,107,66,117,116,116,111,110,32,61,32,40,123,32,100,105,115,97,98,108,101,100,44,32,111,110,67,108,105,99,107,32,125,41,32,61,62,32,123,13,10,32,32,32,32,114,101,116,117,114,110,32,40,13,10,32,32,32,32,32,32,60,66,117,116,116,111,110,13,10,32,32,32,32,32,32,32,32,100,105,115,97,98,108,101,100,61,123,100,105,115,97,98,108,101,100,125,13,10,32,32,32,32,32,32,32,32,111,110,67,108,105,99,107,61,123,111,110,67,108,105,99,107,125,13,10,32,32,32,32,32,32,32,32,118,97,114,105,97,110,116,61,34,99,111,110,116,97,105,110,101,100,34,13,10,32,32,32,32,32,32,32,32,99,111,108,111,114,61,34,115,101,99,111,110,100,97,114,121,34,13,10,32,32,32,32,32,32,62,13,10,32,32,32,32,32,32,32,32,66,97,99,107,13,10,32,32,32,32,32,32,60,47,66,117,116,116,111,110,62,13,10,32,32,32,32,41,59,13,10,32,32,125,59,13,10,32,32,13,10,32,32,101,120,112,111,114,116,32,100,101,102,97,117,108,116,32,66,97,99,107,66,117,116,116,111,110,59,10,10,10,10,10,10,47,47,32,104,101,121,32,103,117,121,115,32,97,108,115,111,32,100,111,32,115,111,109,101,32,98,97,115,105,99,32,99,115,115,32,102,111,114,32,116,104,101,32,98,117,116,116,111,110,44,32,106,117,115,116,32,115,111,32,119,101,32,104,97,118,101,32,97,32,115,101,110,115,101,32,111,102,32,108,97,121,111,117,116,46,32,78,111,116,104,105,110,103,32,99,114,97,122,121,32,40,119,105,100,116,104,44,32,104,101,105,103,104,116,44,32,98,111,114,100,101,114,32,99,111,108,111,114,41