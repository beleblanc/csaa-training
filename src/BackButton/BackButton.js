import React from 'react';
import { useWizardContext} from '../Wizard/Wizard';

const BackButton = () => {
    const { currentPageNumber, goToPreviousPage } = useWizardContext();

    
    return (currentPageNumber !==1 )? 
      (<button
        type='button'
        onClick={goToPreviousPage}
      >
        Back
      </button>): null;

    
  };
  
  export default BackButton;