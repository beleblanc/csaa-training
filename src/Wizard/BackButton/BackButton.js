import React from "react";
import { useWizardContext } from "../Wizard.hooks";

const BackButton = () => {
  const { currentPageNumber, goToPreviousPage } = useWizardContext();

  return currentPageNumber !== 1 ? (
    <button type="button" onClick={goToPreviousPage}>
      Back
    </button>
  ) : null;
};

export default BackButton;
