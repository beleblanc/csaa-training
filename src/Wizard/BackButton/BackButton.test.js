import React from "react";
import { screen, render } from "@testing-library/react";
import BackButton from "./BackButton";

const mockReturnedValued = { currentPageNumber: 4 };
jest.mock("../Wizard.hooks", () => {
  return {
    useWizardContext: () => mockReturnedValued,
  };
});

describe("BackButton", () => {
  it("should render the back button", () => {
    const { container } = render(<BackButton>Back</BackButton>);
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Back")).toBeInTheDocument();
  });

  it("should not render the back button when is in the first page", () => {
    mockReturnedValued.currentPageNumber = 1;
    render(<BackButton>Back</BackButton>);

    expect(screen.queryByText("Back")).not.toBeInTheDocument();
  });
});
