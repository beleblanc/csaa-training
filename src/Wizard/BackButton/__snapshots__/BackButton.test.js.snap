// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render the back button 1`] = `
<div>
  <button
    type="button"
  >
    Back
  </button>
</div>
`;
